src/java/org/apache/cassandra/index/sai/plan/QueryView.java

```diff
@@ -111,11 +111,24 @@ static class Builder
         static class MissingIndexException extends RuntimeException
         {
             final IndexContext context;
+            final Object dataObject;
 
-            public MissingIndexException(IndexContext context, String message)
+            public MissingIndexException(IndexContext context, Object dataObject)
```

Comment:
Nit: passing an `Object`, when we actually only expect specific classes in practice, is always a bit dodgy (nothing strongly breaks if we pass a random object, but the error message, which gets to users, wouldn't make any sense). Granted, the class is only used locally and is package protected, so it's not a big deal, but this kind of thing can change overtime. Fwiw, I would at least keep `dataObject` private, and either 1) keep that ctor private and add 2 public ones, one for `Memtable` and one for `SSTableReader`, or 2) at least have an assertion in the ctor that the object is one of those types.

Reply:
Agreed, it is dodgy, and I don't like it. 
Making it private and adding two static factory methods will be a better, less fragile design. 
Ił change it! Thanks for the idea!